platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/EventDeduplicatorWiring.java

```diff
@@ -30,8 +32,8 @@
  * @param flushRunnable                    the runnable to flush the deduplicator
  */
 public record EventDeduplicatorWiring(
-        @NonNull InputWire<GossipEvent, GossipEvent> eventInput,
-        @NonNull InputWire<Long, GossipEvent> minimumGenerationNonAncientInput,
+        @NonNull BindableInputWire<GossipEvent, GossipEvent> eventInput,
```

Comment:
Hmm... this is perhaps an unfortunate side effect of using records. The goal of creating an `InputWire<IN>` was so that this interface didn't need to expose the output type. The output type of each input wire is really an implementation detail of the binding.

Two possible ways around this.

1) turn these back into classes, and make the getter methods return `InputWire` instead of `BindableInputWire`.
2) store these as `InputWire<>` types, and cast to the bindable type at binding time. Ugly, but it makes the API cleaner.

Perhaps we can discuss this with Lazar tomorrow.